jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -76,23 +85,37 @@ def __init__(self, *args, **kwargs):
         # TODO [markers]: existing markers need converting
         self.spectrum_viewer.state.add_callback('x_display_unit',
                                                 self._on_glue_x_display_unit_changed)
+
         self.spectrum_viewer.state.add_callback('y_display_unit',
                                                 self._on_glue_y_display_unit_changed)
 
         self.spectral_unit = UnitSelectPluginComponent(self,
                                                        items='spectral_unit_items',
                                                        selected='spectral_unit_selected')
-        self.flux_or_sb_unit = UnitSelectPluginComponent(self,
-                                                         items='flux_unit_items',
-                                                         selected='flux_unit_selected')
+
         self.flux_or_sb = SelectPluginComponent(self,
                                                 items='flux_or_sb_items',
                                                 selected='flux_or_sb_selected',
                                                 manual_options=['Surface Brightness', 'Flux'])
 
+        self.flux_unit = UnitSelectPluginComponent(self,
+                                                   items='flux_unit_items',
+                                                   selected='flux_unit_selected')
+
+        self.sb_unit = UnitSelectPluginComponent(self,
+                                                 items='sb_unit_items',
+                                                 selected='sb_unit_selected')
+
     @property
     def user_api(self):
-        return PluginUserApi(self, expose=('spectral_unit', 'flux_or_sb', 'flux_or_sb_unit'))
+        if self.app.config == 'cubeviz':
+            return PluginUserApi(self, expose=('spectral_unit', 'flux_or_sb', 'flux_unit', 'sb_unit'))   # noqa
+        elif self.app.config == 'specviz' and not self.specviz_disabler:
+            return PluginUserApi(self, expose=('spectral_unit', 'flux_unit', 'sb_unit'))
+        if self.specviz_disabler == 'Flux':
+            return PluginUserApi(self, expose=('spectral_unit', 'sb_unit'))
+        if self.specviz_disabler == 'Surface Brightness':
+            return PluginUserApi(self, expose=('spectral_unit', 'flux_unit'))
```

Comment:
these would then apply to all other configs, right (mosviz, imviz, etc)?  The name `specviz_disabler` really seems a bit confusing if that's the case.

Reply:
I will make sure to generalize the name more. I did see this comment just now, I will make sure to bring this up at tagup: https://github.com/spacetelescope/jdaviz/blob/f5b1f482d1514b3076b23a723e9efc1a9a03c7d9/jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py#L77

Reply:
that should be out of date now, we have support for that with the att_display_unit in glue for contours and stretch, right?